### DISCUSSION TITLE
Getting input and output device without starting the midiManager

### DISCUSSION THREAD

**Discussion by elmcapp** [orchetect/swift-midi#131] [Q&A] [answered]
In order to get a list of midi input and output devices is it require to first start the midiManager? Currently when I just fetch for a list of inputs and outputs I get empty array unless I start the manager. I just wanted to know if that is the correct behavior

**Comment by orchetect** [comment, marked as answer]
Yes at present that is intended behavior - the manager maintains endpoint caching behavior internally. Generally all MIDIKit functionality is meant to be available after starting the manager.

There is next to no overhead to start the manager however. Is there a particular use case where you can't start it or don't want to start it?
